lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,10 +71,16 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
-          if ActiveModel::Serializer.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
+          if serializer.class._type.nil?
+            if ActiveModel::Serializer.config.type_source == :model
+              type = serializer.object.class.name.underscore
```

Comment:
The problem is that weird conversion of namespace comes from `plural` or `singular` methods. I Thought it would be reasonable to rely on general pluralization applied to string. But I can roll back to previous behavior of course.